src/dmd/id.d

```diff
@@ -342,6 +342,7 @@ immutable Msgtable[] msgtable =
     { "__equals"},
     { "__switch"},
     { "__switch_error"},
+    { "___d_newitemT" },
```

Comment:
No need for triple underscore prefix and `T` suffix:

```suggestion
    { "__newitem" },
```

Reply:
I also consider double underscore to be wrong usage for regular identifiers in dmd, as these are reserved by the language specification.

There are a couple of places where an identifier behaves slightly different if it starts with `__`, e.g. https://github.com/dlang/dmd/blob/master/src/dmd/traits.d#L1392.

The mago debugger can also hide variables that start with a double underscore as these are usually compiler generated temporaries.

Reply:
I agree with @rainers - the function are `extern (D)` and templates non the less, so we don't need to resort to legacy C-style naming practices. I think for the time being, we should at least use a name consistent with `__switch_error` and `__equals`, like `__new_item`, though `newHeapObject` (or something along those lines) would be better, considering that we probably want to lower all `new` expressions to it.

Reply:
I think the problem is these symbols are in object.d which is automatically imported by all .d source files.  This could cause a naming conflict in user code.  If we could fully qualify the symbol (e.g. `object.switchError` instead of just `switchError`) in the lowering, we wouldn't need the double underscore prefix.  I don't know how to do that.  @WalterBright, do you?

Reply:
Although the use of double underscores is probably unnecessary in `object.d`, too, I wasn't actually referring to their use in lowerings, but their use in dmd as `Id.__newitem`. That way a reserved identifier used in lowerings appears in regular code aswell.

Reply:
>  If we could fully qualify the symbol (e.g. object.switchError instead of just switchError) in the lowering,

Sure we can and have done that even in the past ;-)
It looks like this:

https://github.com/dlang/dmd/blob/fd8eb477ee0553a0e9aaaa8284c1d1d1e299e616/src/dmd/expressionsem.d#L5663-L5666

Reply:
> If we could fully qualify the symbol

That should be a requirement. I've already run into issues with a similar case where the fully qualified name wasn't used.

Reply:
Regarding the double underscore. Since these need to be public symbols, the double underscore could be used to indicate that these symbols should not be used directly, only by the compiler.